content/company/strategy/code-graph/batch-changes/index.md

```diff
@@ -46,6 +46,12 @@ To deliver our vision, we will:
 - Then, release Batch Changes on Sourcegraph Cloud, and test and learn how we can serve individuals and smaller teams.
 - Eventually, address the problem end to end by becoming the go-to place for code change tools and recipes.
 
+## Competitive landscape
+
+Our primary competition comes from companies building internal tools to apply and track code changes. Individual users inside companies also build scripts around [git-xargs](https://github.com/gruntwork-io/git-xargs) or [turbolift](https://github.com/Skyscanner/turbolift) to create changesets, then tend to outgrow it when they scale usage (lack of changeset tracking, collaboration, reproducibility). Every tool relies on a search engine to return matches, so being integrated with Sourcegraph search is a key competitive advantage for Batch Changes.
+
+We expect to see more vertical competition from companies focused on a language ecosystem, that enter the market by building refactoring frameworks for a given language ecosystem, then competing with Batch Changes on tooling to apply and track code changes. [Moderne](https://moderne.io/)/[openrewrite](https://github.com/openrewrite/rewrite) is the first significant example of this, focused on the Java ecosystem. We are committed to remaining language-agnostic, which includes using Batch Changes to apply and track changesets created with openrewrite or any other refactoring framework. We also think there is an opportunity in leveraging precise code intelligence to create radically better code change tools that support many languages in the future.
```

Comment:
@macraig do you think there's an update on your side regarding this? Or is it really more that Batch Changes will be a customer of the features you drive, and there aren't integration considerations into how you're thinking about prioritization right now?

Reply:
@jyavorska making it so other teams like Batch Changes can leverage our metadata is part of our mission statement. We haven't prioritized any specific work on that front, but it's something that we always keep in mind to make sure we're building for a future "Code intel platform" that will have Batch Changes, Search and Code Insights as our internal customers.